client/web/compose/src/views/Admin/Pages/Builder.vue

```diff
@@ -272,6 +276,7 @@ export default {
       page: undefined,
       blocks: [],
       board: null,
+      hasUpdatedBlocks: false,
```

Comment:
Not like this.

1. Create a Set, that will hold indexes of changed blocks
2. if it has any items that means blocks have been modified/added
3. On block save or move the blockIndex(this index wont change if you move blocks around) of that block should be added to the set
4. You can always check if set contains that index very easily, to determine if a certain block has been modified
3. Reset that set on save